packages/zql-benchmarks/src/zbugs.bench.ts

```diff
@@ -0,0 +1,134 @@
+import {Database} from '../../zqlite/src/db.ts';
+import {newQueryDelegate} from '../../zqlite/src/test/source-factory.ts';
+import {schema, builder} from './schema.ts';
+import {testLogConfig} from '../../otel/src/test-log-config.ts';
+import {bench, run, summary} from 'mitata';
+
+const dbPath = process.env.ZBUGS_REPLICA_PATH;
+
+if (!dbPath) {
+  console.error(
+    'Cannot run zbugs.bench.ts without a path to the zbugs replica. Set env var: `ZBUGS_REPLICA_PATH`',
+  );
+  process.exit(0);
+}
+
+// Open the zbugs SQLite database
+const db = new Database(createSilentLogContext(), dbPath);
+const lc = createSilentLogContext();
+
+// Run ANALYZE to populate SQLite statistics for cost model
+db.exec('ANALYZE;');
+
+// Get table specs using computeZqlSpecs
+const tableSpecs = new Map<string, LiteAndZqlSpec>();
+computeZqlSpecs(createSilentLogContext(), db, tableSpecs);
+
+// Create SQLite cost model
+// const costModel = createSQLiteCostModel(db, tableSpecs);
+// const clientToServerMapper = clientToServer(schema.tables);
+// const serverToClientMapper = serverToClient(schema.tables);
+
+// Create SQLite delegate
+const delegate = newQueryDelegate(lc, testLogConfig, db, schema);
+
+// Helper to set flip to false in all correlated subquery conditions
+function setFlipToFalse(condition: Condition): Condition {
+  if (condition.type === 'correlatedSubquery') {
+    return {
+      ...condition,
+      flip: false,
+      related: {
+        ...condition.related,
+        subquery: setFlipToFalseInAST(condition.related.subquery),
+      },
+    };
+  } else if (condition.type === 'and' || condition.type === 'or') {
+    return {
+      ...condition,
+      conditions: condition.conditions.map(setFlipToFalse),
+    };
+  }
+  return condition;
+}
+
+function setFlipToFalseInAST(ast: AST): AST {
+  return {
+    ...ast,
+    where: ast.where ? setFlipToFalse(ast.where) : undefined,
+    related: ast.related?.map(r => ({
+      ...r,
+      subquery: setFlipToFalseInAST(r.subquery),
+    })),
+  };
+}
+
+// Helper to create a query from an AST
+function createQuery<TTable extends keyof typeof schema.tables & string>(
+  tableName: TTable,
+  queryAST: AST,
+) {
+  return new QueryImpl(
+    schema,
+    tableName,
+    queryAST,
+    defaultFormat,
+    'test',
+    undefined,
+    undefined,
+  );
+}
+
+// Helper to benchmark planned vs unplanned
+function benchmarkQuery<TTable extends keyof typeof schema.tables & string>(
+  name: string,
+  query: AnyQuery,
+) {
+  const unplannedAST = asQueryInternals(query).ast;
+
+  // const mappedAST = mapAST(unplannedAST, clientToServerMapper);
+  // const mappedASTCopy = setFlipToFalseInAST(mappedAST);
+  // const dbg = new AccumulatorDebugger();
+  // const plannedServerAST = planQuery(mappedASTCopy, costModel, dbg);
+  // const plannedClientAST = mapAST(plannedServerAST, serverToClientMapper);
+  // const plannedQuery = createQuery(tableName, plannedClientAST);
```

Comment:
I suppose we can remove all this commented out code. Was useful for me when comparing planned and unplanned variants.